pennylane_lightning/core/src/simulators/lightning_gpu/catalyst/LightningGPUSimulator.hpp

```diff
@@ -0,0 +1,181 @@
+// Copyright 2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file LightningGPUSimulator.hpp
+ */
+
+#pragma once
+
+#define __device_lightning_gpu
+
+#include <bitset>
+#include <cmath>
+#include <cstdint>
+#include <iostream>
+#include <limits>
+#include <optional>
+#include <random>
+#include <span>
+#include <stdexcept>
+#include <unordered_map>
+
+#include "AdjointJacobianGPU.hpp"
+#include "MeasurementsGPU.hpp"
+#include "StateVectorCudaManaged.hpp"
+
+#include "CacheManager.hpp"
+#include "Exception.hpp"
+#include "LightningGPUObsManager.hpp"
+#include "QuantumDevice.hpp"
+#include "QubitManager.hpp"
+#include "Utils.hpp"
```

Comment:
It would be great if you took this opportunity while the details are fresh in your mind to document the function and methods used from as many includes as possible. We don't have this for LKokkos because this was more like moving code.

Reply:
Let's revisit it later in a follow-up PR once SV backends have feature parity/